include/Transport.hpp

```diff
@@ -120,13 +140,17 @@ namespace mediasoupclient
 		  const nlohmann::json* codecOptions,
 		  const nlohmann::json& appData = nlohmann::json::object());
 
+		DataProducer* ProduceData(DataProducer::Listener* listener,
```

Comment:
In libmediasoupclient we don't use `ProducerOptions` `ConsumerOptions` but the constructor takes the needed arguments separately.

Lets build the `DataChannelInit` instance internally on constructor with the given arguments. There is no need to expose it to the user in this case.